src/main/java/de/tum/cit/aet/artemis/core/security/ArtemisAuthenticationEventListener.java

```diff
@@ -0,0 +1,102 @@
+package de.tum.cit.aet.artemis.core.security;
+
+import static de.tum.cit.aet.artemis.core.config.Constants.PROFILE_CORE;
+
+import java.net.URL;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.annotation.Profile;
+import org.springframework.security.authentication.event.AuthenticationSuccessEvent;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Component;
+
+import de.tum.cit.aet.artemis.communication.service.notifications.MailSendingService;
+import de.tum.cit.aet.artemis.core.domain.User;
+import de.tum.cit.aet.artemis.core.exception.EntityNotFoundException;
+import de.tum.cit.aet.artemis.core.repository.UserRepository;
+
+/**
+ * Listener for successful authentication events in the Artemis system.
+ * This component listens for successful login attempts and sends email notifications
+ * to non-internal users when they successfully log in.
+ */
+@Profile(PROFILE_CORE)
+@Component
```

Comment:
```suggestion
/**
 * Listener for successful authentication events in the Artemis system.
 * This component listens for successful login attempts and sends email notifications
 * to users when they successfully log in.
 */
@Profile(PROFILE_CORE)
@Component
```

I would probably also use singular (users vs user) for the javadoc, but no strong opinion here. I just stumbled a little across the `users` as it sounded to me that it would send out multiple mails on one login (which is not the case?!) - But really a nitpick and opinionated, just something I noticed, not required to be changed for my approve